### DISCUSSION TITLE
Simulate loading state?

### DISCUSSION THREAD

**Discussion by YCMitch** [vercel/next.js#51234] [Help]
### Summary

I have a `loading.jsx` file in my routes to display a loading state when loading a new route segment, but given each one only loads for ~1 second or so, it's really hard to style it.

Is there a way to force the loading state to be active, so I can lay it out properly?

**Comment by devjiwonchoi** [comment]
Since the `loading.js` is built on top of the React [Suspense](https://react.dev/reference/react/Suspense), you can use the `React Developer Tools` in order to manually toggle the `Suspense boundaries`.

> While designing loading UI, you may find it helpful to use the [React Developer Tools](https://react.dev/learn/react-developer-tools) to manually toggle Suspense boundaries.

![Screenshot 2023-06-13 at 10 40 22 PM](https://github.com/vercel/next.js/assets/120007119/87447a37-dfa7-4d3c-9fe4-79f094fcf56a)

## Demo:

https://github.com/vercel/next.js/assets/120007119/d3b7f423-eae4-4360-a237-c7b04a813f41

## Reference
- [Loading.js](https://nextjs.org/docs/app/api-reference/file-conventions/loading)
- [React Suspense](https://react.dev/reference/react/Suspense)
- [React Developer Tools](https://react.dev/learn/react-developer-tools)

**Reply by Augustine-edeh** [reply]
> Since the `loading.js` is built on top of the React [Suspense](https://react.dev/reference/react/Suspense), you can use the `React Developer Tools` in order to manually toggle the `Suspense boundaries`.
> 
> > While designing loading UI, you may find it helpful to use the [React Developer Tools](https://react.dev/learn/react-developer-tools) to manually toggle Suspense boundaries.
> 
> ![Screenshot 2023-06-13 at 10 40 22 PM](https://private-user-images.githubusercontent.com/120007119/245495651-87447a37-dfa7-4d3c-9fe4-79f094fcf56a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMTAxMTMsIm5iZiI6MTczNjIwOTgxMywicGF0aCI6Ii8xMjAwMDcxMTkvMjQ1NDk1NjUxLTg3NDQ3YTM3LWRmYTctNGQzYy05ZmU0LTc5ZjA5NGZjZjU2YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwN1QwMDMwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDA1NWVjZTBjOGQwYmQzYjIyOGQ2MzBhNDU3ZTcwZDNjN2M0Mzg1ZjQ1MmRhMjE0NDM4Njk3NzcwNjQ1ZDZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.j7TtMbx2OayarQ96E8S_qv9zHTUakLmtGWnBt7uqB1E)
> 
> ## Demo:
>  Screen.Recording.2023-06-13.at.10.46.44.PM.mov 
> ## Reference
> * [Loading.js](https://nextjs.org/docs/app/api-reference/file-conventions/loading)
> * [React Suspense](https://react.dev/reference/react/Suspense)
> * [React Developer Tools](https://react.dev/learn/react-developer-tools)

Thanks for the help. Found this really helpful!
